app/courses/[courseId]/quiz/[quizId]/student.tsx

```diff
@@ -0,0 +1,212 @@
+  }, [currentIndex]);
+
+  // Fetch quiz data
+  const { contentsByIds, currentUserInfo } = useLazyLoadQuery<studentQuizQuery>(
+    graphql`
+      query studentQuizQuery($id: [UUID!]!) {
+        currentUserInfo {
+          id
+        }
+        contentsByIds(ids: $id) {
+          id
+          metadata {
+            name
+          }
+          ... on QuizAssessment {
+            quiz {
+              assessmentId
+              questionPool {
+                ... on MultipleChoiceQuestion {
+                  text
+                  answers {
+                    text
+                    correct
+                    feedback
+                  }
+                  numberOfCorrectAnswers
+                }
+                id
+                number
+                type
+                hint
+              }
+              requiredCorrectAnswers
+              questionPoolingMode
+              numberOfRandomlySelectedQuestions
+              selectedQuestions {
+                ... on MultipleChoiceQuestion {
+                  text
+                  answers {
+                    text
+                    correct
+                    feedback
+                  }
+                  numberOfCorrectAnswers
+                }
+                id
+                number
+                type
+                hint
+              }
+            }
+          }
+        }
+      }
+    `,
+    { id: [quizId] }
+  );
+
+  const quiz = contentsByIds[0].quiz;
+  const currentQuestion = quiz!.selectedQuestions[currentIndex];
+  const questionText = currentQuestion.text;
+  const answers = currentQuestion.answers;
+
+  const nextQuestion = async () => {
+    if (!checkAnswers) {
+      setCheckAnswers(true);
+    } else if (
+      checkAnswers &&
+      currentIndex + 1 < (quiz?.selectedQuestions.length ?? 0)
+    ) {
+      setCheckAnswers(false);
+      setCurrentIndex(currentIndex + 1);
+    } else {
+      router.push(`/courses/${courseId}`);
+    }
+  };
+
+  const handleAnswerChange = (index: any) => {
+    setUserAnswers((prevAnswers) => {
+      const newAnswers = [...prevAnswers];
+      newAnswers[currentIndex] = index;
+      return newAnswers;
+    });
+  };
+
+  return (
+    <main>
+      <Heading title={contentsByIds[0].metadata.name} backButton />
+      <InfoDialog
+        open={infoDialogOpen}
+        onClose={() => setInfoDialogOpen(false)}
+        title={questionText!}
+        hint={currentQuestion.hint!}
+      />
+      <div className="w-full border-b border-b-gray-300 mt-6 flex justify-center">
+        <div className="bg-white -mb-[9px] px-3 text-xs text-gray-600">
+          {currentIndex + 1}/{quiz?.selectedQuestions.length ?? 0}
+        </div>
+      </div>
+
+      <div className="mt-6 text-center text-gray-600">{questionText}</div>
+
+      <div className="w-full border-b border-b-gray-300 mt-6 flex justify-center mb-6">
+        <div>
+          <IconButton onClick={() => setInfoDialogOpen(true)}>
```

Comment:
Not sure, but a text button with "Hints" would be clearer here? Or maybe a tooltip

Reply:
Done as button